From 0a8883d96540a9bce0e3791fa612f1569231dba1 Mon Sep 17 00:00:00 2001
From: Brian Bennett <brian.bennett@joyent.com>
Date: Wed, 4 Jan 2017 10:18:17 -0800
Subject: [PATCH] NET-368 NAPI resolvers should not be shuffled for booting CNs
 Reviewed by: Cody Mello <melloc@joyent.com> Reviewed by: Richard Kiene
 <richard.kiene@joyent.com>

---
 lib/bootparams.js | 20 +-------------------
 1 file changed, 1 insertion(+), 19 deletions(-)

diff --git a/lib/bootparams.js b/lib/bootparams.js
index 33c619f..47e01ba 100644
--- a/lib/bootparams.js
+++ b/lib/bootparams.js
@@ -97,24 +97,6 @@ function getFromCache(opts, callback) {
 }
 
 
-/**
- * Shuffle the contents of the array based on the Fisher-Yates or Knuth shuffle.
- */
-function shuffle(array) {
-    var m = array.length, t, i;
-
-    while (m) {
-        i = Math.floor(Math.random() * m--);
-        t = array[m];
-        array[m] = array[i];
-        array[i] = t;
-    }
-
-    return array;
-}
-
-
-
 // --- Exported functions
 
 
@@ -321,7 +303,7 @@ function getBootParams(opts, callback) {
 
         params.ip = bootNic.ip;
         params.netmask = bootNic.netmask;
-        params.resolvers = shuffle(bootNic.resolvers);
+        params.resolvers = bootNic.resolvers;
 
         // Allow kernel_args from CNAPI to override the nic tag values, but
         // dutifully complain about it
-- 
2.21.0

